File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionCoalesceTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionCoalesceTest extends BaseTest {
    @Test
    void testObjectFunctionCoalesce() {
        var arr = new String[] { null, null, "c"};
        assertTransformation(arr, fromJson("""
{
  "$$coalesce": "$"
}
"""), arr[2]);
        assertTransformation(arr, fromJson("""
{
  "$$coalesce": ["$[0]", "b", "c"]
}
"""), "b");
        // alias
        assertTransformation(arr, fromJson("""
{
  "$$first": "$"
}
"""), arr[2]);
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionCoalesce.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionCoalesce", () => {
  test("object", async () => {
    const arr = [null, null, "c"];
    await assertTransformation(arr, { $$coalesce: "$" }, arr[2]);
    await assertTransformation(arr, { $$coalesce: ["$[0]", "b", "c"] }, "b");
  });

  test("aliasFirst", async () => {
    const arr = [null, null, "c"];
    await assertTransformation(arr, { $$first: "$" }, arr[2]);
  });
});
